File: 7_Function_Overriding.py
# Runtime Polymorphism -> Dynamic Binding | Overriding | Late Binding

class Animal:
    def speak(self):
        return "Animal Sound"
    def food(self):
        return "Animal Food"

class Dog(Animal):
    def speak(self):
        return "Woof!"
    def food(self):
        return "Eat Meat"
class Cat(Animal):
    def speak(self):
        return "Meow!"
    def food(self):
        return "Drink Milk"
def animals(animal):
    print(animal.speak())
    print(animal.food())
dog = Dog()
cat = Cat()
animals(dog)
animals(cat)

File: 7_Function_Overriding.cpp
#include <bits/stdc++.h>
using namespace std;
class Animal {
public:
    virtual string speak() {
        return "Animal Sound";
    }
    virtual string food() {
        return "Animal Food";
    }
};
class Dog : public Animal {
public:
   string speak() override {
        return "Woof!";
    }
    string food() override {
        return "Eat Meat";
    }
};

class Cat : public Animal {
public:
    string speak() override {
        return "Meow!";
    }
    string food() override {
        return "Drink Milk";
    }
};
void animals(Animal* animal) {
    cout << animal->speak() << endl;
    cout << animal->food() << endl;
}
int main(){

    Dog dog;   
    Cat cat;   
    animals(&dog); 
    animals(&cat); 

    return 0;

}



